qek/data/extractors.py

```diff
@@ -745,13 +773,33 @@ def __init__(
     def batch_ids(self) -> list[str] | None:
         return self._batch_ids
 
-    def run(self) -> CloudExtracted:
+    @abc.abstractmethod
+    def run(
+        self,
+    ) -> PasqalCloudExtracted:
+        """
+        Launch the extraction.
+        """
+        raise Exception("Not implemented")
+
+    def _run(
+        self,
+        state_extractor: Callable[[Job, pl.Sequence], dict[str, int] | None],
+        emulator: EmulatorType | None,
+        config: BaseConfig | None,
+    ) -> PasqalCloudExtracted:
         if len(self.sequences) == 0:
             logger.warning("No sequences to run, did you forget to call compile()?")
```

Comment:
Why just logging a warning and letting it through ?

Reply:
What would you suggest? Raising an exception?

Reply:
Idk, do we continue if there isn't anything to execute ? In some sense, we can't prevent users from running empty jobs.